From 002688a371fcaf01da1c7712a73650304884441c Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Wed, 9 Jul 2014 15:03:44 +0300
Subject: [PATCH 0003/1345] fix: xor: added shutdown capability to disable
 clocks

commit  3876bf5c2a8b60cf6d6ca3d791b701c416bbef44 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/9097
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/9142
Signed-off-by: Neta Zur <neta@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/16147

Conflicts:
	drivers/dma/mv_xor.c

Change-Id: I476be491dfffc1eff5a5a90dee0e617b4c0f9a0b
Reviewed-on: http://vgitil04.il.marvell.com:8080/27274
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/mv_xor.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/dma/mv_xor.c b/drivers/dma/mv_xor.c
index 81214e7..aed82b3 100644
--- a/drivers/dma/mv_xor.c
+++ b/drivers/dma/mv_xor.c
@@ -1615,10 +1615,21 @@ static int mv_xor_probe(struct platform_device *pdev)
 	return ret;
 }
 
+static void mv_xor_shutdown(struct platform_device *pdev)
+{
+	struct mv_xor_device *xordev = platform_get_drvdata(pdev);
+
+	if (!IS_ERR(xordev->clk)) {
+		clk_disable_unprepare(xordev->clk);
+		clk_put(xordev->clk);
+	}
+}
+
 static struct platform_driver mv_xor_driver = {
 	.probe		= mv_xor_probe,
 	.suspend        = mv_xor_suspend,
 	.resume         = mv_xor_resume,
+	.shutdown	= mv_xor_shutdown,
 	.driver		= {
 		.name	        = MV_XOR_NAME,
 		.of_match_table = of_match_ptr(mv_xor_dt_ids),
-- 
1.7.9.5

